tests/unit/useNewTransactionsTest.ts

```diff
@@ -0,0 +1,294 @@
+        expect(result.current).toEqual([]);
+
+        // 2. Report is loaded and transactions data is not loaded yet
+        rerender({
+            hasOnceLoadedReportActions: true,
+            transactions: [],
+        });
+        await delay(10);
+        expect(result.current).toEqual([]);
+
+        // 3. Report is loaded and transactions data is loaded
+        rerender({
+            hasOnceLoadedReportActions: true,
+            transactions: transactionsAlreadyInReport,
+        });
+        expect(result.current).toEqual([]); // there is no new transactions, because the transactions that were already in the report are not considered new
+    });
+
+    it('returns new transactions when transactions are added after initial load', async () => {
+        // 1. Report and transactions data is not loaded yet
+        const {rerender, result} = renderHook((props) => useNewTransactions(props.hasOnceLoadedReportActions, props.transactions), {
+            initialProps: {
+                hasOnceLoadedReportActions: false,
+                transactions: [] as Transaction[],
+            },
+        });
+        await delay(10); // We need to wait to ensure that usePrevious hook updates correctly
+        expect(result.current).toEqual([]);
+
+        // 2. Report is loaded and transactions data is not loaded yet
+        rerender({
+            hasOnceLoadedReportActions: true,
+            transactions: [],
+        });
+        await delay(10);
+        expect(result.current).toEqual([]);
+
+        // 3. Report is loaded and transactions data is loaded
+        rerender({
+            hasOnceLoadedReportActions: true,
+            transactions: transactionsAlreadyInReport,
+        });
+        await delay(10);
+        expect(result.current).toEqual([]);
+
+        // 4. User added new transaction
+        rerender({
+            hasOnceLoadedReportActions: true,
+            transactions: [...transactionsAlreadyInReport, newTransaction],
+        });
+        expect(result.current).toEqual([newTransaction]);
+    });
+
+    it('returns new transactions when adding transactions to empty report', async () => {
+        // 1. Report and transactions data is not loaded yet
+        const {rerender, result} = renderHook((props) => useNewTransactions(props.hasOnceLoadedReportActions, props.transactions), {
+            initialProps: {
+                hasOnceLoadedReportActions: false,
+                transactions: [] as Transaction[],
+            },
+        });
+        await delay(10); // We need to wait to ensure that usePrevious hook updates correctly
+
+        // 2. Report is loaded and it has no transactions so there are no further rerenders
+        rerender({
+            hasOnceLoadedReportActions: true,
+            transactions: [],
+        });
+        await delay(1000); // We need to wait to ensure that the second useEffect is executed
```

Comment:
"the second" useEffect is not very informative.

IMO either got the simpler route and write
`// We need to wait to ensure that all effects are executed`
OR got he more specific/descriptive route and write something like:
`// [...] wait to ensure that that the useEffect performing XXX work is executed`